Library/Homebrew/utils/pypi.rb

```diff
@@ -159,30 +228,18 @@ def self.update_python_resources!(formula, version: nil, package_name: nil, extr
     main_package = if package_name.present?
       Package.new(package_name)
     else
-      begin
-        Package.new(formula.stable.url, is_url: true)
-      rescue ArgumentError
-        nil
-      end
-    end
-
-    if main_package.blank?
-      return if ignore_non_pypi_packages
-
-      odie <<~EOS
-        Could not infer PyPI package name from URL:
-          #{Formatter.url(formula.stable.url)}
-      EOS
+      Package.new(formula.stable.url, is_url: true)
     end
 
-    unless main_package.valid_pypi_package?
-      return if ignore_non_pypi_packages
-
-      odie "\"#{main_package}\" is not available on PyPI."
+    if version.present?
+      if main_package.valid_pypi_package?
+        main_package.version = version
+      else
+        return if ignore_non_pypi_packages
```

Comment:
This is the only change between this PR and #15511: some callers like `bump-formula-pr` expect to be able to call `update_python_resources!` on arbitrary formulae and have it exit silently.